pkg/errors/cancel.go

```diff
@@ -0,0 +1,27 @@
+package errors
+
+import (
+	"context"
+	stderrors "errors"
+
+	"github.com/cloudquery/cq-provider-sdk/provider/diag"
+
+	gcodes "google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func IsCancelation(err error) bool {
```

Comment:
Cancelation or Cancellation? 🤔 Both is correct but since `context.Canceled` is with one L I wanted to keep it the same.